bolt/CMakeLists.txt

```diff
@@ -82,7 +82,7 @@ endforeach()
 
 set(BOLT_ENABLE_RUNTIME_default OFF)
 if ((CMAKE_SYSTEM_PROCESSOR STREQUAL "x86_64"
-    OR CMAKE_SYSTEM_PROCESSOR STREQUAL "aarch64")
+    OR CMAKE_SYSTEM_PROCESSOR MATCHES "^(armv[8-9].*|arm64|aarch64)$")
```

Comment:
I'm not clear that this can ever be equal to 'armv*', do you have reason to believe it can be? Looking at the docs [0], and some discussion from [1], I conclude this could be equal to aarch64 or arm64 across the different platforms. That said I don't currently see the harm of somewhat generous matching here if you want to keep it or make it more broad. Maybe consider `^(arm|aarch64)` to allow various suffices maybe add a comment that it's intended to at least match armv8-a, armv9-a, arm64, aarch64, if it's believed to be necessary.

[0] https://cmake.org/cmake/help/latest/variable/CMAKE_HOST_SYSTEM_PROCESSOR.html
[1] https://stackoverflow.com/questions/70475665/what-are-the-possible-values-of-cmake-system-processor

Reply:
Thanks for your investigation @peterwaller-arm ! I'm not too deep in this field and I agree that most likely arm64/aarch64 would be enough, I just thought that armv8 - armv9 check won't hurt anyone - that is why I include them. Also I remember such checks in other  projects I saw. As for arm check - I'm not convinced it is need, not only I didn't see it anywhere, but it also might give a wrong idea/false positive check for 32 bit ARM CPUs, so I'm currently not convinced we need to include such check here..

Reply:
👍, I suggest then to match arm64 and aarch64 for now, and add others if they are seen in practice.

Reply:
Well I don't really mind. At the end of the day to really handle all possibilities we need also to check for upper cases situations or AArch64, so let's stick to aarch64 and arm64 for now